### PR TITLE
New context selector

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-1052

### What?
This adds the new context selector table.

NOTE: Two of the requirements on the tickets has not been implemented, since they are related to other tickets and it's better not to implement those now.

Date selector and Country selector we'll be moved to a different page.

### Why?
to improve UI

### CODE REVIEW

File: app/helpers/commodities_helper.rb [hunk 1]
```diff
@@ -33,6 +33,16 @@ def commodity_tree(main_commodity, commodities)
     end
   end
 
+  def classification_description(commodity)
```
Comment (willfish) [inline app/helpers/commodities_helper.rb]: In a declarable helper?

File: app/helpers/declarable_helper.rb [hunk 2]
```diff
@@ -1,4 +1,14 @@
 module DeclarableHelper
+  def classification_description(commodity)
```
Comment (willfish) [inline app/helpers/declarable_helper.rb:2]: We should add a test for this guy

File: app/models/concerns/declarable.rb [hunk 3]
```diff
@@ -57,6 +57,22 @@ def all_footnotes
     ).uniq(&:code)
   end
 
+  def supplementary_unit(country)
```
Comment (willfish) [inline app/models/concerns/declarable.rb]: I think we need a spec for this guy

File: app/models/measure_component.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class MeasureComponent
+  include ApiEntity
+
+  has_one :duty_expression # (p/st)
```
Comment (willfish) [inline app/models/measure_component.rb]: We don't need these comments

File: app/presenters/declarable_presenter.rb [hunk 5]
```diff
@@ -8,11 +8,11 @@ def initialize(declarable)
   end
 
   def to_s
-    declarable.formatted_description
+    declarable.formatted_description.html_safe
```
Comment (willfish) [inline app/presenters/declarable_presenter.rb]: I wonder if we should sanitize :thinking:

File: app/views/headings/show.html.erb [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
 
 <% content_for :bodyClasses, "page--wide" %>
 
+  <%= page_header heading_for(section: @section,chapter: @chapter,heading: @heading, commodity: @commodity) %>
+
+  <%= render 'shared/search/date_picker_form' %>
+  <%= render 'shared/context_tables/heading' %>
+  <%= render 'shared/callouts/heading'%>
+
+<!-- TODO: what's the new UI when the heading is declarable ? -->
```
Comment (willfish) [inline app/views/headings/show.html.erb]: Remove me

File: app/views/headings/show.html.erb [hunk 7]
```diff
@@ -37,5 +41,7 @@
     </div>
     <%= render 'footnote', footnotes: @heading.all_footnotes %>
   </article>
-  <%= render 'shared/notes', section_note: @heading.section.section_note, chapter_note: @heading.chapter.chapter_note %>
 <% end %>
+
+<%= render 'shared/notes', section_note: @heading.section.section_note, chapter_note: @heading.chapter.chapter_note %>
```
Comment (willfish) [inline app/views/headings/show.html.erb]: Is this correct?

File: app/views/shared/context_tables/_commodity.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<table class="govuk-table">
+  <tbody class="govuk-table__body">
+    <tr class="govuk-table__row ">
```
Comment (willfish) [inline app/views/shared/context_tables/_commodity.html.erb:3]: I feel like we've all fallen into the trap of treating declarable headings and commodities differently when they should be treated exactly the same. I'm wondering if we should rename this to _declarable.html.erb and just render the declarable or the non-declarable version of this partial.

I'm not picky about it as this is an idiom we're not always using

File: app/views/shared/search/_date_picker_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<%= form_tag perform_search_path, method: :get, class: "tariff-search #{@section_css}", id: "new_search" do |f| %>
+  <%= yield %>
```
Comment (willfish) [inline app/views/shared/search/_date_picker_form.html.erb:2]: What are we yielding?

File: app/helpers/declarable_helper.rb (hunk 2 above)
Comment (alexdesi) [reply app/helpers/declarable_helper.rb:2]: 👍

**Review by willfish — APPROVED** [review]

**Pull request merged**